src/sparkx/Histogram.py

```diff
@@ -268,12 +271,25 @@ def add_value(self, value):
         ----------
         value: int, float, np.number, list, numpy.ndarray
             Value(s) which are supposed to be added to the histogram instance.
+        weight: int, float, np.number, list, numpy.ndarray, optional
+            Weight(s) associated with the value(s). If provided, it should have 
+            the same length as the value parameter.
 
         Raises
         ------
         TypeError
             if the input is not a number or numpy.ndarray or list
+        ValueError
+            if the input `weight` has not the same dimension as `value`
         """
+        # Check if weight has the same length as value
+        if weight is not None:
```

Comment:
Should we use `None` or `nan`?  What happens if I use particle properties as weights?

Reply:
We will use None and build in a nan check for the weight value.